### ISSUE TITLE
Can't use Typescript types due to wrong packaging

### ISSUE THREAD

**Issue by michaelarnauts** [bpmn-io/form-js#529]
### Describe the Bug

I'm trying to use form-js in an Angular project, but it seems that the files aren't correctly packaged. 

I'm seeing the following issues:

```
...
Error: ../node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:127:31 - error TS2307: Cannot find module './types' or its corresponding type declarations.

127 export type Injector = import('./types').Injector;
                                  ~~~~~~~~~


Error: ../node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:128:27 - error TS2307: Cannot find module './types' or its corresponding type declarations.

128 export type Data = import('./types').Data;
                              ~~~~~~~~~


Error: ../node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:129:29 - error TS2307: Cannot find module './types' or its corresponding type declarations.

129 export type Errors = import('./types').Errors;
                                ~~~~~~~~~


Error: ../node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:130:29 - error TS2307: Cannot find module './types' or its corresponding type declarations.

130 export type Schema = import('./types').Schema;
                                ~~~~~~~~~
...
```

Part of Form.d.ts
```
...
export type Injector = import('./types').Injector;
export type Data = import('./types').Data;
export type Errors = import('./types').Errors;
export type Schema = import('./types').Schema;
export type FormProperties = import('./types').FormProperties;
export type FormProperty = import('./types').FormProperty;
export type FormEvent = import('./types').FormEvent;
export type FormOptions = import('./types').FormOptions;
...
```

So it seems that the file `@bpmn-io/form-js-viewer/dist/types/Form.d.ts` is trying to read the file `types` in the current directory, but the actual `types.d.ts` file is still in the `src/` subfolder.

```
.
├── dist
│   ├── assets
│   │   ├── flatpickr
│   │   │   └── light.css
│   │   └── form-js.css
│   ├── index.cjs
│   ├── index.cjs.map
│   ├── index.es.js
│   ├── index.es.js.map
│   └── types
│       ├── core
│       │   └── ...
│       ├── import
│       │   └── ...
│       ├── render
│       │   └── ...
│       ├── src
│       │   └── types.d.ts
│       └── util
│       │   └── ...
│       ├── Form.d.ts
│       ├── index.d.ts
├── LICENSE
├── package.json
└── README.md
```

### Steps to Reproduce

1. npm install @bpmn-io/form-js-viewer
2. Add `import {Form} from '@bpmn-io/form-js-viewer';` on a typescript file.
2. Try to compile in an typescript project
3. It's complaining that the file `types` can't be found.

I think it's related to https://github.com/bpmn-io/form-js/pull/143.

### Environment

- Host: Node v16.17.1 / Angular 12
- OS: Ubuntu 22.04
- Library version: 0.11.0

**Comment by michaelarnauts** [comment]
# Minimal steps to reproduce


```ts
# index.ts
import { Form } from '@bpmn-io/form-js-viewer';
```


```
$ cd /tmp/
$ mkdir formjs-test/
$ cd formjs-test/
$ npm install @bpmn-io/form-js-viewer typescript
$ echo "import { Form } from '@bpmn-io/form-js-viewer';" > index.ts
$ cat index.ts
import { Form } from '@bpmn-io/form-js-viewer';
$ node_modules/.bin/tsc index.ts
node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:127:31 - error TS2307: Cannot find module './types' or its corresponding type declarations.

127 export type Injector = import('./types').Injector;
                                  ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:128:27 - error TS2307: Cannot find module './types' or its corresponding type declarations.

128 export type Data = import('./types').Data;
                              ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:129:29 - error TS2307: Cannot find module './types' or its corresponding type declarations.

129 export type Errors = import('./types').Errors;
                                ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:130:29 - error TS2307: Cannot find module './types' or its corresponding type declarations.

130 export type Schema = import('./types').Schema;
                                ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:131:37 - error TS2307: Cannot find module './types' or its corresponding type declarations.

131 export type FormProperties = import('./types').FormProperties;
                                        ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:132:35 - error TS2307: Cannot find module './types' or its corresponding type declarations.

132 export type FormProperty = import('./types').FormProperty;
                                      ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:133:32 - error TS2307: Cannot find module './types' or its corresponding type declarations.

133 export type FormEvent = import('./types').FormEvent;
                                   ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:134:34 - error TS2307: Cannot find module './types' or its corresponding type declarations.

134 export type FormOptions = import('./types').FormOptions;
                                     ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:142:13 - error TS2456: Type alias 'OnEventWithPriority' circularly references itself.

142 export type OnEventWithPriority = OnEventWithPriority;
                ~~~~~~~~~~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:143:13 - error TS2456: Type alias 'OnEventWithOutPriority' circularly references itself.

143 export type OnEventWithOutPriority = OnEventWithOutPriority;
                ~~~~~~~~~~~~~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/core/ConditionChecker.d.ts:30:44 - error TS2307: Cannot find module '../types' or its corresponding type declarations.

30     check(condition: string, data?: import('../types').Data): boolean | null;
                                              ~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/core/ConditionChecker.d.ts:49:48 - error TS2307: Cannot find module '../types' or its corresponding type declarations.

49     evaluate(expression: string, data?: import('../types').Data): any;
                                                  ~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/index.d.ts:15:40 - error TS2307: Cannot find module './types' or its corresponding type declarations.

15 export type CreateFormOptions = import('./types').CreateFormOptions;
                                          ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:2:14 - error TS2304: Cannot find name 'ButtonIcon'.

2     export { ButtonIcon as button };
               ~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:3:14 - error TS2304: Cannot find name 'CheckboxIcon'.

3     export { CheckboxIcon as checkbox };
               ~~~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:4:14 - error TS2304: Cannot find name 'ChecklistIcon'.

4     export { ChecklistIcon as checklist };
               ~~~~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:5:14 - error TS2304: Cannot find name 'ColumnsIcon'.

5     export { ColumnsIcon as columns };
               ~~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:6:14 - error TS2304: Cannot find name 'DatetimeIcon'.

6     export { DatetimeIcon as datetime };
               ~~~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:7:14 - error TS2304: Cannot find name 'ImageIcon'.

7     export { ImageIcon as image };
               ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:8:14 - error TS2304: Cannot find name 'NumberIcon'.

8     export { NumberIcon as number };
               ~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:9:14 - error TS2304: Cannot find name 'RadioIcon'.

9     export { RadioIcon as radio };
               ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:10:14 - error TS2304: Cannot find name 'SelectIcon'.

10     export { SelectIcon as select };
                ~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:10:14 - error TS2552: Cannot find name 'SelectIcon'. Did you mean 'Selection'?

10     export { SelectIcon as select };
                ~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:13621:13
    13621 declare var Selection: {
                      ~~~~~~~~~
    'Selection' is declared here.

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:11:14 - error TS2304: Cannot find name 'TaglistIcon'.

11     export { TaglistIcon as taglist };
                ~~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:12:14 - error TS2304: Cannot find name 'TextIcon'.

12     export { TextIcon as text };
                ~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:13:14 - error TS2304: Cannot find name 'TextfieldIcon'.

13     export { TextfieldIcon as textfield };
                ~~~~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:14:14 - error TS2304: Cannot find name 'TextareaIcon'.

14     export { TextareaIcon as textarea };
                ~~~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:15:14 - error TS2304: Cannot find name 'FormIcon'.

15     export { FormIcon as default };
                ~~~~~~~~


Found 28 errors in 4 files.

Errors  Files
    10  node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:127
     2  node_modules/@bpmn-io/form-js-viewer/dist/types/core/ConditionChecker.d.ts:30
     1  node_modules/@bpmn-io/form-js-viewer/dist/types/index.d.ts:15
    15  node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:2
```

**Comment by pinussilvestrus** [comment]
Thanks for the detailed steps! It seems like we accidentally moved the types with `@0.10.0`. I will investigate a fix for that.

**Comment by michaelarnauts** [comment]
I've just tried with 0.9, but I still see the circularly references there. The type file seems to be in the correct place.

```
$ npm install @bpmn-io/form-js-viewer@0.9 typescript
$ node_modules/.bin/tsc index.ts
node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:138:13 - error TS2456: Type alias 'OnEventWithPriority' circularly references itself.

138 export type OnEventWithPriority = OnEventWithPriority;
                ~~~~~~~~~~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:139:13 - error TS2456: Type alias 'OnEventWithOutPriority' circularly references itself.

139 export type OnEventWithOutPriority = OnEventWithOutPriority;
                ~~~~~~~~~~~~~~~~~~~~~~


Found 2 errors in the same file, starting at: node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:138
```

**Comment by pinussilvestrus** [comment]
For the time being, would using the `--skipLibCheck` flag be a possible workaround for you?

**Comment by pinussilvestrus** [comment]
I drafted a fix here: https://github.com/bpmn-io/form-js/pull/530 (not yet finished).

**Comment by shadouuw** [comment]
> I've just tried with 0.9, but I still see the circularly references there. The type file seems to be in the correct place.
> 
> ```
> $ npm install @bpmn-io/form-js-viewer@0.9 typescript
> $ node_modules/.bin/tsc index.ts
> node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:138:13 - error TS2456: Type alias 'OnEventWithPriority' circularly references itself.
> 
> 138 export type OnEventWithPriority = OnEventWithPriority;
>                 ~~~~~~~~~~~~~~~~~~~
> 
> node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:139:13 - error TS2456: Type alias 'OnEventWithOutPriority' circularly references itself.
> 
> 139 export type OnEventWithOutPriority = OnEventWithOutPriority;
>                 ~~~~~~~~~~~~~~~~~~~~~~
> 
> 
> Found 2 errors in the same file, starting at: node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:138
> ```

you just need to replace those line with this it should fix it : 
export type OnEventWithPriority = (type: FormEvent, priority: number, handler: Function) => void;
export type  OnEventWithOutPriority = (type: FormEvent, handler: Function) => void;
export type OnEventType = OnEventWithPriority & OnEventWithOutPriority;

**Comment by michaelarnauts** [comment]
> For the time being, would using the `--skipLibCheck` flag be a possible workaround for you?

This works, but I really want to turn the check back on, so I hope it's still a priority to fix this :)

**Comment by pinussilvestrus** [comment]
Fixed via https://github.com/bpmn-io/form-js/commit/bda9710520387cd730195c1a25b439a762a61b29, will be released via `0.12.1` (probably end of this week)
